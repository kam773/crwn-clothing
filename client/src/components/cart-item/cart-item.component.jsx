import React from 'react';

import {
  CartItemContainer,
  ItemDetailsContainer,
  CartItemImage
} from './cart-item.styles';

const CartItem = (props) => {
  const {imageUrl, name, price, quantity} = props
  return (
    <CartItemContainer>
        <CartItemImage src={imageUrl} alt='item' />
        <ItemDetailsContainer>
          <span className="name">{name}</span>
          <span className="price">
            {quantity} x ${price}
          </span>
        </ItemDetailsContainer>
      </CartItemContainer>
  )
}


export default React.memo(CartItem);