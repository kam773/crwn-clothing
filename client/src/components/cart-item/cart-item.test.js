import React from 'react';
import { shallow } from 'enzyme';
import CartItem from './cart-item.component';
import { CartItemImage } from './cart-item.styles';
import renderer from 'react-test-renderer';

const mockProps = {
  imageUrl: 'url',
  price: 20,
  name: 'name',
  quantity: 2,
};

describe('create snapshot', () => {
  it('capture snapshot of CartItem', () => {
    const wrapper = renderer
      .create(<CartItem {...mockProps} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('shallow render react component', () => {
  it('render the DUMB component', () => {
    const wrapper = shallow(<CartItem {...mockProps} />);
    expect(wrapper.length).toEqual(1);
  });
  it('render image', () => {
    const wrapper = shallow(<CartItem {...mockProps} />);
    expect(wrapper.find(CartItemImage)).toBeDefined();
  });
  it('render price and name', () => {
    const wrapper = shallow(<CartItem {...mockProps} />);
    expect(wrapper.find('span').length).toEqual(2);
  });
});
