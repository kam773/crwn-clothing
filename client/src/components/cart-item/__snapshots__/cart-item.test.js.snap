// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot capture snapshot of CartItem 1`] = `
<div
  className="sc-bdVaJa eTYnIE"
>
  <img
    alt="item"
    className="sc-bwzfXH hTyRPc"
    src="url"
  />
  <div
    className="sc-htpNat kvpPYA"
  >
    <span
      className="name"
    >
      name
    </span>
    <span
      className="price"
    >
      2
       x $
      20
    </span>
  </div>
</div>
`;
